### DISCUSSION TITLE
Will and when kong use OpenSSL v3.0 ?

### DISCUSSION THREAD

**Discussion by lihongbj** [Kong/kong#9446] [Help] [answered]
Now kong OSS and Enterprise use OpenSSL version `1.1.1`, and it is noticed that per https://konghq.com/blog/kong-gateway-2-7,  BoringSSL will be introduced to replace OpenSSL for FIPS feature in kong Enterprise. But per BoringSSL [home site](https://github.com/boringssl/boringssl), `Although BoringSSL is an open source project, it is not intended for general use, as OpenSSL is. We don't recommend that third parties depend upon it. Doing so is likely to be frustrating because there are no guarantees of API or ABI stability.`, the BoringSSL solution seems a temporary workaround  when OpenSSL does not support FIPS . But now the latest OpenSSL has declared it supports FIPS since `3.0.x`. so will kong adopt OpenSSL 3.0 in the future ?

**Comment by fffonion** [comment, marked as answer]
BoringSSL is now used by Kong 3.0 FIPS package. However whether for FIPS or not, Kong will support OpenSSL 3.0 as that's new mainline version, but time is not planned yet. We will move FIPS package to base on OpenSSL 3.0 once the new certification is announced.
